base/sparse/sparsematrix.jl

```diff
@@ -700,7 +700,8 @@ function gen_broadcast_body_sparse(f::Function, is_first_sparse::Bool)
         colptr1 = A_1.colptr; rowval1 = A_1.rowval; nzval1 = A_1.nzval
         colptr2 = A_2.colptr; rowval2 = A_2.rowval; nzval2 = A_2.nzval
 
-        nnzB = nnz(A_1) * div(B.n, A_1.n) * div(B.m, A_1.m)  +
+        nnzB = isempty(B) ? 0 :
```

Comment:
The ternary has very low precedence so it's covering all the way up to `A_2.m`.  Will add parens in an attempt to make it clearer.